src/charm.py

```diff
@@ -9,76 +9,181 @@
 """
 
 import logging
+from typing import Dict
 
-from charms.data_platform_libs.v0.database_requires import (
+from charms.data_platform_libs.v0.data_interfaces import (
     DatabaseCreatedEvent,
     DatabaseRequires,
+    KafkaRequires,
+    TopicCreatedEvent,
 )
 from ops.charm import ActionEvent, CharmBase
 from ops.main import main
 from ops.model import ActiveStatus, BlockedStatus
 
 logger = logging.getLogger(__name__)
 
+KAFKA_EXTRA_USER_ROLES = "consumer,producer"
+
 
 class IntegratorCharm(CharmBase):
     """Integrator charm that connects to database charms."""
 
     def __init__(self, *args):
         super().__init__(*args)
 
-        self.database = self.model.config.get("database", "")
+        self.database = self.model.config.get("database-name", "")
```

Comment:
This is a minor comment but I would allow database to have "None" value if it is not specified. 

I can see that it was implemented with "" before, so it is more of a suggestion up for you to follow up if you agree and if it is not too much of an hassle